syntax = "proto3";


service Jumia {
  rpc GetProduct (ProductRequest) returns (ProductResponse) {}
  rpc GetProducts (ProductRequest) returns (stream ProductResponse) {}
  rpc GetCheapestProduct_Stream (stream ProductRequest) returns (ProductResponse) {}
  rpc GetCheapestProduct_Message (stream ProductRequest) returns (stream ProductResponse) {}
}

message Page {
  int32 id = 1;
}

message Category {
  string link = 1;
  Page page = 2;
}

message ProductRequest {
  Category category = 1;
  int32 index = 2;
}

message ProductResponse {
  string name = 1;
  string link = 2;
  string price = 3;
  string rating = 4;
  string rated_sales = 5;
}
